google/cloud/spanner/connection.h

```diff
@@ -128,6 +128,9 @@ class Connection {
   };
   //@}
 
+  /// Returns any options use by the concrete Connection.
+  virtual Options options() { return Options{}; }
```

Comment:
Done.

Reply:
I don't see a connection test? Only the tests that mock the return value.

I think it would be nice to have a test that checks that we default the supplied options. Maybe something like:
```cc
TEST(ConnectionImplTest, DefaultedOptions) {
  auto conn = MakeConnection(/* db */, Options{}.set<GrpcNumChannelsOption>(42););
  auto opts = conn->options();
  // Verify that supplied options are used.
  EXPECT_EQ(42, opts.get<GrpcNumChannelsOption>());
  // Verify that options are defaulted.
  EXPECT_EQ(internal::DefaultOptions().get<EndpointOption>(), opts.get<EndpointOption>());
}
```

Reply:
`ConnectionImpl` doesn't know anything (any longer) about default options.  Indeed, `MakeConnection()` is in the `Client` API.  So, I added something similar to your suggestion to `client_test.cc`.  PTAL.